ls++

```diff
@@ -362,6 +383,9 @@ sub perm_size_file {
 
 sub perm {
   my ($perm) = @_;
+  if (substr($perm,-1,1) ne "+"){
+    $perm .= " "; # adding an extra space to align with occasional +
```

Comment:
shouldn't take much performance, you don't need to parse the permissions just calculate the length of it without spaces and pad everything shorter then the longest one.
I think it would make sense to only show the space when there is need for it